### PR TITLE
Update FileSystemAclExtensions.cs null checks (#41907)

### PR DESCRIPTION
**Pull request by eanova**
Fix for issue #41907 
Added necessary null checks and exceptions to GetAccessControl and SetAccessControl .  Removed  braces in Create(this FileInfo,..) method.

cc: @carlossanlop @danmosemsft

### CODE REVIEW

**Comment by eanova** [comment]
I'll submit the PR for the new repo when available.

**Comment by danmoseley** [comment]
Thanks, could you please link to the issue?

File: src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs [hunk 1]
```diff
@@ -52,21 +52,27 @@ public static void SetAccessControl(this FileInfo fileInfo, FileSecurity fileSec
 
         public static FileSecurity GetAccessControl(this FileStream fileStream)
         {
+            if (fileStream == null)
+                throw new NullReferenceException(nameof(fileStream));
```
Comment (carlossanlop) [inline src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs:56]: After looking at other extension methods, I noticed we don't use `NullReferenceException` for `this` parameters. We use `ArgumentNullException` instead. Examples:

- [corefx/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.cs](https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchive.Create.cs)

- [corefx/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs](https://github.com/dotnet/corefx/blob/master/src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFileExtensions.ZipArchiveEntry.Extract.cs)

Also, do you mind adding the exception to the triple slash comments?

**Review by carlossanlop — CHANGES_REQUESTED** [review]
Thank you for your contribution, @eanova. I left a couple of requests for you to consider.

File: src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs (hunk 1 above)
Comment (eanova) [reply src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs:56]: Gotcha. This was a source of confusion.  The tests will also need to be updated as they are expecting NullReferenceException.  I'll put all the changes into a PR for the new repo using the ZipArchive extensions as reference.

Thanks!

Comment (carlossanlop) [reply src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs:56]: Sure, if you want to submit the PR in the new repo, that's fine too. Thank you again.

**Comment by carlossanlop** [comment]
FYI: the dotnet/runtime repo has been opened to the public, so feel free to open this PR over there.

**Comment by carlossanlop** [comment]
I'll close this issue. Feel free to create a new PR in dotnet/runtime whenever you're ready and I'll sign it off.

**Pull request closed without merging**